src/uri.cr

```diff
@@ -426,4 +426,23 @@ class URI
       URI.escape(password, io)
     end
   end
+
+  DEFAULT_PORTS = {
+    "ftp"    => 21,
+    "ftps"   => 990,
+    "gopher" => 70,
+    "http"   => 80,
+    "https"  => 443,
+    "ldap"   => 389,
+    "ldaps"  => 636,
+    "nntp"   => 119,
+    "scp"    => 22,
+    "sftp"   => 22,
+    "ssh"    => 22,
+    "telnet" => 23,
+  }
+
+  private def is_default_port?
+    return port.nil? || port == DEFAULT_PORTS[scheme]?
```

Comment:
Just a minor semantic thing: `nil` does not necessarily mean "default port", it might also be that the scheme doesn't support ports at all (for example `file` or `mailto`). Therefore I'd suggest to move the nil check out of this method.